SetFlags.cmake

```diff
@@ -233,87 +81,76 @@ macro(enable_profile)
+		set(CMAKE_CXX_FLAGS_COVERAGE "${CMAKE_CXX_FLAGS_COVERAGE} -D_DEBUG")
+		set(CMAKE_C_FLAGS_COVERAGE   "${CMAKE_C_FLAGS_COVERAGE}   -D_DEBUG")
+		set(CMAKE_CXX_FLAGS_RELEASE  "${CMAKE_CXX_FLAGS_RELEASE}  -DNDEBUG")
+		set(CMAKE_C_FLAGS_RELEASE    "${CMAKE_C_FLAGS_RELEASE}    -DNDEBUG")
+	endif()
 
-		# we support non-IEEE 754 fpus so can make no guarentees about error
-		add_flags_cxx("-ffast-math")
+	# Allow for a forced 32-bit build under 64-bit OS:
+	if (FORCE_32)
+		add_compile_options(-m32)
+		add_link_options(-m32)
+	endif()
 
-		if(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
-			# backtrace() and friends are in libexecinfo
-			add_flags_lnk("-lexecinfo")
-		endif()
+	# Have the compiler generate code specifically targeted at the current machine on Linux:
+	if(LINUX AND NOT NO_NATIVE_OPTIMIZATION)
+		add_compile_options(-march=native)
+	endif()
+endfunction()
 
-		if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-			if ("${CLANG_VERSION}" VERSION_LESS 3.0)
-				message(FATAL_ERROR "Cuberite requires clang version 3.0 or higher, your version is ${CLANG_VERSION}")
-			endif()
-			# clang does not provide the __extern_always_inline macro and a part of libm depends on this when using fast-math
-			add_flags_cxx("-D__extern_always_inline=inline")
-			add_flags_cxx("-Weverything -Werror -Wno-c++98-compat-pedantic -Wno-string-conversion")
-			add_flags_cxx("-Wno-exit-time-destructors -Wno-padded -Wno-weak-vtables")
-			add_flags_cxx("-Wno-switch-enum")  # This is a pretty useless warning, we've already got -Wswitch which is what we need
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.0)
-				# flags that are not present in 3.0
-				add_flags_cxx("-Wno-implicit-fallthrough")
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.1)
-				# flags introduced in 3.2
-				add_flags_cxx("-Wno-documentation")
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.4)
-				add_flags_cxx("-Wno-error=disabled-macro-expansion")
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.5)
-				include(CheckCXXCompilerFlag)
-				check_cxx_compiler_flag(-Wno-reserved-id-macro HAS_NO_RESERVED_ID_MACRO)
-				check_cxx_compiler_flag(-Wno-documentation-unknown-command HAS_NO_DOCUMENTATION_UNKNOWN)
-				if (HAS_NO_RESERVED_ID_MACRO)
-					# Use this flag to ignore error for a reserved macro problem in sqlite 3
-					add_flags_cxx("-Wno-reserved-id-macro")
-				endif()
-				if (HAS_NO_DOCUMENTATION_UNKNOWN)
-					# Ignore another problem in sqlite
-					add_flags_cxx("-Wno-documentation-unknown-command")
-				endif()
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.7)
-				check_cxx_compiler_flag(-Wno-double-promotion HAS_NO_DOUBLE_PROMOTION)
-				if (HAS_NO_DOUBLE_PROMOTION)
-					add_flags_cxx("-Wno-double-promotion")
-				endif()
-			endif()
-			add_flags_cxx("-Wno-error=unused-command-line-argument")
-			add_flags_cxx("-Wno-documentation-unknown-command")
-		endif()
+function(enable_warnings TARGET)
```

Comment:
`set_exe_flags` made much more sense. `-fsigned-char` is definitely not a warning.